ext/storage_sources/s3_store/test/test_s3_connection.cpp

```diff
@@ -24,15 +27,215 @@ int TestListBuckets(const Aws::S3Crt::ClientConfiguration &config);
 int
 TestListBuckets(const Aws::S3Crt::ClientConfiguration &config)
 {
+    int ret = 0;
     S3Connection conn(config);
     std::vector<std::string> buckets;
-    if (!conn.ListBuckets(buckets))
-        return 1;
+    if (ret = conn.ListBuckets(buckets) != 0)
+        goto err;
 
     std::cout << "All buckets under my account:" << std::endl;
     for (const auto &bucket : buckets)
         std::cout << "  * " << bucket << std::endl;
-    return 0;
+
+err:
+    return ret;
+}
+
+/*
+ * TestListObjects --
+ *     Unit test for listing S3 objects under the first bucket in the associated AWS account. This
+ *     test assumes there are initially no objects with the prefix of "test_list_objects_" in the
+ *     bucket.
+ */
+int
+TestListObjects(const Aws::S3Crt::ClientConfiguration &config)
+{
+    int ret = 0;
+    S3Connection conn(config);
+
+    /* Temporary workaround to get a bucket to use. */
+    std::string firstBucket;
+    std::vector<std::string> buckets;
+
+    if (ret = conn.ListBuckets(buckets) != 0)
+        goto err;
```

Comment:
This test has over 20 `goto` statements. Please work with someone to rewrite this differently.